lib/TickingFixedClock.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lendable\Clock;
+
+/**
+ * Simple in-memory ticking fixed clock.
+ *
+ * The original fixed value will advance forward as normal to simulate time passing from
+ * the fixed time point.
+ */
+final class TickingFixedClock implements MutableClock
+{
+    private \DateTimeImmutable $now;
+
+    private function __construct(\DateTimeInterface $now, private \DateTimeImmutable $tickingFrom)
+    {
+        $this->now = \DateTimeImmutable::createFromInterface($now);
+    }
+
+    public static function tickingFromCurrentTime(\DateTimeImmutable $now): self
+    {
+        return self::tickingFromTime($now, new \DateTimeImmutable('now'));
+    }
+
+    /**
+     * Constructs with a specific point in time the clock was originally began ticking from.
+     *
+     * This allows for testing where you are persisting state of a clock across in-memory boundaries
+     * and need to re-hydrate into that state, rather than begin ticking time from the current time again.
+     */
+    public static function tickingFromTime(\DateTimeImmutable $now, \DateTimeImmutable $tickingFrom): self
```

Comment:
Not done in this PR: a `PersistedTickingFixedClock` counterpart, which this method provides support for.